### ISSUE TITLE
[BUG] koordlet MemoryQoS panic for a LSE pod

### ISSUE THREAD

**Issue by ls-2018** [koordinator-sh/koordinator#1901]
**What happened**:
<img width="1451" alt="image" src="https://github.com/koordinator-sh/koordinator/assets/38821215/0e2472c7-c9f9-4356-ab53-f661bc3698db">

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
```
apiVersion: scheduling.koordinator.sh/v1alpha1
kind: Reservation
metadata:
  name: reservation-demo
  labels:
    app: demo
spec:
  template: # set resource requirements
    namespace: default
    spec:
      containers:
        - args:
            - '-c'
            - '1'
          command:
            - stress
          image: polinux/stress
          imagePullPolicy: Always
          name: stress
          resources: # reserve 500m cpu and 800Mi memory
            requests:
              cpu: 1
              memory: 800Mi
      schedulerName: koord-scheduler # use koord-scheduler
  owners: # set the owner specifications
    - object: # owner pods whose name is `default/pod-demo-0`
        name: pod-demo-0
        namespace: default
  ttl: 1h # set the TTL, the reservation will get expired 1 hour later

#  kubectl get reservation reservation-demo -o wide

```
```
apiVersion: v1
kind: Pod
metadata:
  name: pod-demo-0 # match the owner spec of `reservation-demo`
  labels:
    "koordinator.sh/qosClass": "LSE"
    "koordinator.sh/priority-class": "koord-prod"
  annotations:
#    "scheduling.koordinator.sh/reserve-pod": "true"
    "scheduling.koordinator.sh/reservation-affinity": '{"reservationSelector": {"app": "demo"}}'
#    "scheduling.koordinator.sh/reservation-name": "reservation-demo"

#    "scheduling.koordinator.sh/reservation-order": ""
#    "scheduling.koordinator.sh/reservation-allocated": ""

spec:
  containers:
    - args:
        - '-c'
        - '1'
      command:
        - stress
      image: polinux/stress
      imagePullPolicy: Always
      name: stress
      resources:
        requests:
          cpu: 1
          memory: 800Mi
  restartPolicy: Always
  schedulerName: koord-scheduler # use koord-scheduler

```
**Anything else we need to know?**:

**Environment**:
- App version:  1.3
- Kubernetes version (use `kubectl version`):
 1.24.15
- Install details (e.g. helm install args):
- Node environment (for koordlet/runtime-proxy issue):
  - Containerd/Docker version:
  - OS version:
  - Kernal version:
  - Cgroup driver: cgroupfs/systemd
- Others:

**Comment by saintube** [comment]
@ls-2018 Hi, this issue duplicates #1670. It should be fixed by #1663 and released in the koordlet [v1.4.0](https://github.com/koordinator-sh/koordinator/releases/tag/v1.4.0).

**Comment by ls-2018** [comment]
👌🏻, thanks
